### PR TITLE
User can update the profile once logged in

### PR DESCRIPTION
**Pull request by kianaditya**
[PT Link](https://www.pivotaltracker.com/story/show/162393872)

* User can see his profile on view profile page
* User can update profile on edit profile page
* User can see view profile and edit profile buttons on navbar

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class UsersController < ApplicationController
+    def show
+        @user = User.find(params[:id])
+    end
+    def update
+        @user = User.find(params[:id])
+        # @user.update(user_params)
```
Comment (johannaanden) [inline app/controllers/users_controller.rb]: Remove commented-out code

File: app/models/user.rb [hunk 2]
```diff
@@ -3,4 +3,6 @@ class User < ApplicationRecord
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
```
Comment (johannaanden) [inline app/models/user.rb]: Remove commented-out code

**Review by johannaanden — CHANGES_REQUESTED** [review]

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (kianaditya) [reply app/controllers/users_controller.rb]: Done

File: app/models/user.rb (hunk 2 above)
Comment (kianaditya) [reply app/models/user.rb]: done

File: app/views/partials/_nav.html.haml [hunk 3]
```diff
@@ -2,6 +2,8 @@
 -if user_signed_in?
     %p Hello, #{current_user.email}
     =link_to "Logout", destroy_user_session_path, method: :delete
+    =link_to 'Edit profile', edit_user_registration_path
```
Comment (oliverochman) [inline app/views/partials/_nav.html.haml]: Is it not better to have the "Edit profile" link on the "View Profile" page ?

File: app/views/users/show.html.haml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+- attr = ['firstname', 'lastname', 'address', 'about', 'drivingage', 'age', 'phone', 'email']
```
Comment (oliverochman) [inline app/views/users/show.html.haml]: redo this one <3

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+class UsersController < ApplicationController
+    def show
+        @user = User.find(params[:id])
+    end
+    def update
```
Comment (oliverochman) [inline app/controllers/users_controller.rb]: remove the update action, you dont use it

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+        @user = User.find(params[:id])
+        redirect_to user_path(current_user)
+    end
+
+    private
+    def user_params
```
Comment (oliverochman) [inline app/controllers/users_controller.rb]: this one aswell

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: app/views/partials/_nav.html.haml (hunk 3 above)
Comment (GergKllai1) [reply app/views/partials/_nav.html.haml]: done

File: app/views/users/show.html.haml (hunk 4 above)
Comment (GergKllai1) [reply app/views/users/show.html.haml]: done

File: app/controllers/users_controller.rb (hunk 5 above)
Comment (GergKllai1) [reply app/controllers/users_controller.rb]: done

File: app/controllers/users_controller.rb (hunk 6 above)
Comment (GergKllai1) [reply app/controllers/users_controller.rb]: done

File: app/models/user.rb [hunk 7]
```diff
@@ -1,6 +1,6 @@
 class User < ApplicationRecord
-  # Include default devise modules. Others available are:
-  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
+
+  validates_presence_of :firstname, :lastname, :address, :drivingage, :age
```
Comment (oliverochman) [inline app/models/user.rb]: I want you to change the attributes `firstname, lastname and drivingage` to `driving_age` etc

File: app/views/users/show.html.haml [hunk 8]
```diff
@@ -0,0 +1,27 @@
+=link_to 'Edit profile', edit_user_registration_path
+
+- attr = ['firstname', 'lastname', 'address', 'about', 'drivingage', 'age', 'phone', 'email']
```
Comment (oliverochman) [inline app/views/users/show.html.haml]: you guys dont use this

File: app/views/users/show.html.haml [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+%h4 Name:
+%p= "#{current_user.firstname} #{current_user.lastname}"
+
+%h4 Email:
+%p= current_user.email
+
+%h4 Address:
+%p= current_user.address
```
Comment (oliverochman) [inline app/views/users/show.html.haml]: Using `current_user` here is not a good choice. What if a car owner wants to inspect someone's profile before they decide on renting their car out to him/her.

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: app/models/user.rb (hunk 7 above)
Comment (GergKllai1) [reply app/models/user.rb]: done

File: app/views/users/show.html.haml (hunk 8 above)
Comment (GergKllai1) [reply app/views/users/show.html.haml]: done

File: app/views/users/show.html.haml (hunk 9 above)
Comment (GergKllai1) [reply app/views/users/show.html.haml]: done

**Review by oliverochman — APPROVED** [review]

**Pull request merged**